File: scripts/recolor.js
const sharp = require('sharp');
const path = require('path');

async function recolorImage() {
    const inputPath = path.join(__dirname, '../public/curve.avif');
    const outputPath = path.join(__dirname, '../public/curve.png'); // write to png first

    try {
        console.log(`Loading image from ${inputPath}`);

        // Load image to raw buffer
        const { data, info } = await sharp(inputPath)
            .raw()
            .toBuffer({ resolveWithObject: true });

        console.log(`Loaded ${info.width}x${info.height} with ${info.channels} channels.`);

        // Loop through pixels
        for (let i = 0; i < data.length; i += info.channels) {
            const r = data[i];
            const g = data[i + 1];
            const b = data[i + 2];
            const a = info.channels === 4 ? data[i + 3] : 255;

            // Similar heuristic: 
            if (r > 150 && g < 150 && b < 100 && a > 0) {
                // Orange -> #2D5016 (45, 80, 22)
                data[i] = 45;
                data[i + 1] = 80;
                data[i + 2] = 22;
            } else if (r > 200 && g > 150 && b > 150 && a > 0) {
                // Light pale fill area -> #EBF2EE (235, 242, 238) roughly matches
                data[i] = 235;
                data[i + 1] = 242;
                data[i + 2] = 238;
            }
        }

        // Write the new raw buffer out as AVIF
        await sharp(data, {
            raw: {
                width: info.width,
                height: info.height,
                channels: info.channels
            }
        })
            .avif({ effort: 6 }) // Convert back to AVIF efficiently
            .toFile(path.join(__dirname, '../public/curve_green.avif'));

        console.log("Success! Saved as /public/curve_green.avif");

    } catch (err) {
        console.error("Error recoloring image:", err);
    }
}

recolorImage();

File: scripts/recolor.py
import sys
from PIL import Image

def recolor_image(input_path, output_path):
    # Open image and convert to RGBA
    img = Image.open(input_path).convert('RGBA')
    data = img.getdata()
    
    # Target color (#FF5A36 or similar orange hues)
    # the target is roughly red=255, green=90, blue=54
    # The replacement color is Forest Green (#2D5016 / 45, 80, 22)
    
    new_data = []
    for item in data:
        r, g, b, a = item
        
        # Simple heuristic to find the orange-red hues. 
        # Typically R is high, G is med/low, B is low.
        if r > 150 and g < 150 and b < 100 and a > 0:
            # We found a pixel that is part of the orange curve
            # Replace it with Forest Green, preserving alpha
            new_data.append((45, 80, 22, a))
            
        elif r > 200 and g > 200 and b > 200:
            # Keep white/transparent background pixels alone
            new_data.append(item)
            
        elif r > 200 and g > 150 and b > 150 and a > 0:
             # This is for the light pale orange fill areas inside the curve
             # We want to map this to a pale green instead. 
             # Let's say light green: #EBF2EE (235, 242, 238) roughly matches
             new_data.append((235, 242, 238, a))
             
        else:
            # Keep all other colored pixels unchanged (e.g. gray text)
            new_data.append(item)
            
    img.putdata(new_data)
    img.save(output_path, "PNG")

if __name__ == "__main__":
    if len(sys.argv) != 3:
        print("Usage: python recolor.py <input> <output>")
        sys.exit(1)
    recolor_image(sys.argv[1], sys.argv[2])
